src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -215,15 +215,24 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                 await WriteStringAsync(request.Method.Method, cancellationToken).ConfigureAwait(false);
                 await WriteByteAsync((byte)' ', cancellationToken).ConfigureAwait(false);
 
-                if (_usingProxy)
+                if (request.Method == s_httpConnectMethod)
```

Comment:
I don't think we want to write the hostname from the request Url here.

The request Url is going to define the server we initially connect to.  So using the request Url's hostname as the CONNECT target would be asking the server to establish a connection to itself, which is not what we want.

Instead, we should use the Host header to define the target for CONNECT.

So:
(1) User creates a request with RequestUri = "http://foo.com/".  We treat "foo.com" as the host we initially establish the connection to.  Path/query are ignored.
(2) User sets the Host header on the request to "bar.com:1234".  We treat "bar.com:1234" as the target of the CONNECT.
(3) We send a request that looks like "CONNECT bar.com:1234 HTTP/1.1".  This will include a Host header that looks like "Host: bar.com:1234".  This is how the RFC expects it to work, as I understand.  
(4) Server responds with 200 OK.  Connection tunnel has been established and we expose the raw stream back to the user.
(5) Alternatively, server response with a different response; we expose this to the user as a regular response with response body.

Reply:
ok, I can update it to work like this. Currently, proxy code puts what ever target URI it wants to connect to. I'm working on better tests to nail this down.